src/lib/y2firstboot/dialogs/wsl_product_selection.rb

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "ui/installation_dialog"
+
+Yast.import "UI"
+
+module Y2Firstboot
+  module Dialogs
+    class WSLProductSelection < ::UI::InstallationDialog
+      include Yast::I18n
+
+      attr_reader :product
+
+      attr_reader :wsl_gui_pattern
+
+      def initialize(products, default_product: nil)
+        textdomain "firstboot"
+
+        super()
+        @products = products
+        @default_product = default_product || products.first
+      end
+
+      def next_handler
+        save
+        super
+      end
+
+      protected
+
+      def dialog_title
+        _("Product Selection")
+      end
+
+      def dialog_content
+        items = products.map { |p| item_for(p) }
+
+        HSquash(
+          VBox(
+            RadioButtonGroup(
+              Id(:product_selector),
+              VBox(
+                Left(Label(_("Select the product to use"))),
+                *items
+              )
+            ),
+            VSpacing(2),
+            Left(CheckBox(Id(:wsl_gui_pattern), _("Install WSL GUI pattern"), false))
+          )
+        )
+      end
+
+      # TODO
+      def help_text
+        _("The WSL GUI pattern installs some needed dependencies for " \
```

Comment:
I'd consult the help text with the requestor. Maybe we could be more specific, it is too generic, it does not provide much help. It should tell more clear why you should install the pattern, what are the advantages, etc...

Reply:
I agree, this is only a sample.